src/test/java/uk/gov/ons/census/casesvc/messaging/InvalidAddressReceiverIT.java

```diff
@@ -0,0 +1,119 @@
+
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class InvalidAddressReceiverIT {
+  @Value("${queueconfig.invalid-address-inbound-queue}")
+  private String invalidAddressInboundQueue;
+
+  @Value("${queueconfig.rh-case-queue}")
+  private String rhCaseQueue;
+
+  @Autowired private RabbitQueueHelper rabbitQueueHelper;
+  @Autowired private CaseRepository caseRepository;
+  @Autowired private EventRepository eventRepository;
+
+  @Before
+  @Transactional
+  public void setUp() {
+    rabbitQueueHelper.purgeQueue(invalidAddressInboundQueue);
+    rabbitQueueHelper.purgeQueue(rhCaseQueue);
+    eventRepository.deleteAllInBatch();
+    caseRepository.deleteAllInBatch();
+  }
+
+  @Test
+  public void testHappyPath() throws InterruptedException, IOException {
+    // GIVEN
+    BlockingQueue<String> outboundQueue = rabbitQueueHelper.listen(rhCaseQueue);
+
+    EasyRandom easyRandom = new EasyRandom();
+    Case caze = easyRandom.nextObject(Case.class);
+    caze.setCaseId(UUID.randomUUID());
+    caze.setUacQidLinks(null);
+    caze.setEvents(null);
+    caze.setAddressInvalid(false);
+    caze = caseRepository.saveAndFlush(caze);
+
+    ResponseManagementEvent managementEvent = new ResponseManagementEvent();
+    managementEvent.setEvent(new EventDTO());
+    managementEvent.getEvent().setDateTime(OffsetDateTime.now());
+    managementEvent.getEvent().setChannel("Test channel");
+    managementEvent.getEvent().setSource("Test source");
+    managementEvent.getEvent().setType(EventTypeDTO.ADDRESS_NOT_VALID);
+    managementEvent.setPayload(new PayloadDTO());
+    managementEvent.getPayload().setInvalidAddress(new InvalidAddress());
+    managementEvent.getPayload().getInvalidAddress().setCollectionCase(new CollectionCaseCaseId());
+    managementEvent
+        .getPayload()
+        .getInvalidAddress()
+        .getCollectionCase()
+        .setId(caze.getCaseId().toString());
+
+    String json = convertObjectToJson(managementEvent);
+    Message message =
+        MessageBuilder.withBody(json.getBytes())
+            .setContentType(MessageProperties.CONTENT_TYPE_JSON)
+            .build();
+    rabbitQueueHelper.sendMessage(invalidAddressInboundQueue, message);
+
+    // check the emitted eventDTO
+    ResponseManagementEvent responseManagementEvent =
+        rabbitQueueHelper.checkExpectedMessageReceived(outboundQueue);
+
+    assertThat(responseManagementEvent.getEvent().getType()).isEqualTo(EventTypeDTO.CASE_UPDATED);
+    CollectionCase actualPayloadCase = responseManagementEvent.getPayload().getCollectionCase();
+    assertThat(actualPayloadCase.getId()).isEqualTo(caze.getCaseId().toString());
+    assertThat(actualPayloadCase.getAddressInvalid()).isTrue();
+
+    // check database for log eventDTO
+    List<Event> events = eventRepository.findAll();
```

Comment:
https://github.com/ONSdigital/census-rm-case-processor/blob/597256fad97835b0eb7a770de249d1323bb400bd/src/test/java/uk/gov/ons/census/casesvc/messaging/InvalidAddressReceiverIT.java#L109-L113

would be happier to see more detailed testing of event record (as in QuestionnaireLinkedReceiverIT), contents of event_payload field have been a real pain

Reply:
we've standardised the logging now

Reply:
added one more thing, but there's adequate test coverage tbh